src/Casting/DTOCast.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace WendellAdriel\ValidatedDTO\Casting;
+
+use Illuminate\Validation\ValidationException;
+use Throwable;
+use WendellAdriel\ValidatedDTO\Exceptions\CastException;
+use WendellAdriel\ValidatedDTO\Exceptions\CastTargetException;
+use WendellAdriel\ValidatedDTO\ValidatedDTO;
+
+class DTOCast implements Castable
+{
+    /**
+     * @param  string  $dtoClass
+     */
+    public function __construct(private string $dtoClass)
+    {
+    }
+
+    /**
+     * @param  string  $property
+     * @param  mixed  $value
+     * @return ValidatedDTO
+     *
+     * @throws CastException|CastTargetException|ValidationException
+     */
+    public function cast(string $property, mixed $value): ValidatedDTO
+    {
+        if (is_string($value)) {
+            $value = json_decode($value, true);
+        }
```

Comment:
Again a case for the `ArrayCast`?

Reply:
I can't use it here because this is intended only for JSON strings while the ArrayCast can convert other types into an array